Makefile

```diff
@@ -94,7 +94,7 @@ CLEAN_FILES += ./node_modules/tape
 test: $(NYC) $(FAUCET)
 	$(NPM) run coverage | $(FAUCET)
 
-$(DOCTOC):
+$(DOCTOC): | $(NPM_EXEC)
```

Comment:
This is to avoid `npm install` re-running every time we run `make check|test`

Reply:
Without the ` | $(NPM_EXEC)` it was causing `npm install` everything every time I ran a make task including doctoc. Once it's added, that doesn't happen anymore.

Has more sense now?

Reply:
But `make check|test` don't depend on doctoc do they? I'm missing why a doctoc target would be called for check|test.

Reply:
I think it's related to be part of node modules. Anyway, here is what happens with it removed:

```
thanagar: triton-kbmapi(TRITON-2068) pedropc$ git status
On branch TRITON-2068
Your branch is up to date with 'origin/TRITON-2068'.

nothing to commit, working tree clean
thanagar: triton-kbmapi(TRITON-2068) pedropc$ fg
nvim README.md

[1]+  Stopped                 nvim README.md
thanagar: triton-kbmapi(TRITON-2068) pedropc$ git diff
diff --git a/Makefile b/Makefile
index 84dce3b..81b4aec 100644
--- a/Makefile
+++ b/Makefile
@@ -94,7 +94,7 @@ CLEAN_FILES += ./node_modules/tape
 test: $(NYC) $(FAUCET)
        $(NPM) run coverage | $(FAUCET)

-$(DOCTOC): | $(NPM_EXEC)
+$(DOCTOC):
        $(NPM) install doctoc

 # Make a table of contents in Markdown docs that are setup to use it.  This
thanagar: triton-kbmapi(TRITON-2068) pedropc$ make docs
node_modules/.bin/doctoc --notitle --maxlevel 3 docs/README.md

DocToccing single file "docs/README.md" for github.com.

==================

"docs/README.md" will be updated

Everything is OK.
thanagar: triton-kbmapi(TRITON-2068) pedropc$ make check
xmllint --noout --path deps/eng/tools/ --dtdvalid deps/eng/tools/service_bundle.dtd.1 smf/manifests/kbmapi.xml
xmllint --noout --path deps/eng/tools/ --dtdvalid deps/eng/tools/service_bundle.dtd.1 smf/manifests/kbmtr.xml
npm install --no-save doctoc@^1.3.1 eslint@^4.19.1 eslint-plugin-joyent@^2.1.0 faucet@0.0.1 moray-sandbox@^0.2.1 nyc@^14.1.0 tape@^4.6.0 && \
	    touch node_modules/.bin/eslint
npm WARN deprecated circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.
npm WARN deprecated clone@1.0.2: XSS vulnerability fixed in v1.0.3
npm WARN gentlyRm not removing /Users/pedropc/work/triton-kbmapi/node_modules/.bin/uuid as it wasn't installed by /Users/pedropc/work/triton-kbmapi/node_modules/uuid
- assert-plus@1.0.0 node_modules/fast/node_modules/assert-plus
- bunyan@1.8.12 node_modules/fast/node_modules/bunyan
- verror@1.10.0 node_modules/fast/node_modules/verror
- assert-plus@1.0.0 node_modules/forkexec/node_modules/assert-plus
- verror@1.10.0 node_modules/forkexec/node_modules/verror
- assert-plus@1.0.0 node_modules/kang/node_modules/assert-plus
- backoff@2.5.0 node_modules/kang/node_modules/backoff
- dtrace-provider@0.8.8 node_modules/kang/node_modules/bunyan/node_modules/dtrace-provider
- bunyan@1.8.12 node_modules/kang/node_modules/bunyan
- node-uuid@1.4.8 node_modules/kang/node_modules/node-uuid
- verror@1.10.0 node_modules/kang/node_modules/verror
- assert-plus@1.0.0 node_modules/macaddr/node_modules/assert-plus
- verror@1.10.0 node_modules/macaddr/node_modules/verror
- assert-plus@1.0.0 node_modules/moray-filter/node_modules/assert-plus
- bunyan@1.8.12 node_modules/node-manatee/node_modules/bunyan
- assert-plus@1.0.0 node_modules/node-manatee/node_modules/verror/node_modules/assert-plus
- verror@1.10.0 node_modules/node-manatee/node_modules/verror
uuid@3.0.0 node_modules/uuid -> node_modules/moray-sandbox/node_modules/uuid
kbmapi@1.0.0 /Users/pedropc/work/triton-kbmapi
├── doctoc@1.4.0
├── eslint@4.19.1
├── eslint-plugin-joyent@2.1.0
├── faucet@0.0.1
├─┬ moray-sandbox@0.2.1
│ └─┬ moray-server@2.3.0  (git+https://github.com/joyent/moray.git#74bc4d6289886a15c28c277c6dfb4ec990229809)
│   ├── node-manatee@1.1.0  (git+https://github.com/joyent/node-manatee.git#262828a805463b169916dfb0d725d28dbf40f882)
│   └─┬ restify@5.2.0
│     └── uuid@3.4.0
├── nyc@14.1.1
└── tape@4.13.0

node node_modules/.bin/eslint  lib/apis/moray.js lib/endpoints/pivtokens.js lib/endpoints/index.js lib/endpoints/recovery-configurations.js lib/util/constants.js lib/util/errors.js lib/util/validate.js lib/auth.js lib/models/recovery-token.js lib/models/fsm-transition.js lib/models/recovery-configuration-transition.js lib/models/recovery-configuration.js lib/models/index.js lib/models/pivtoken.js lib/models/pivtoken-history.js lib/models/model.js lib/config.js lib/cli/do_completion.js lib/cli/do_pivtoken/do_add.js lib/cli/do_pivtoken/index.js lib/cli/do_pivtoken/do_remove.js lib/cli/do_pivtoken/do_list.js lib/cli/index.js lib/cli/do_recovery/do_activate.js lib/cli/do_recovery/do_add.js lib/cli/do_recovery/index.js lib/cli/do_recovery/do_remove.js lib/cli/do_recovery/do_get.js lib/cli/do_recovery/do_list.js lib/cli/do_recovery/do_stage.js lib/cli/do_recovery/do_cancel.js lib/cli/common.js lib/app.js client/index.js test/unit/resources/pivtokens.test.js test/unit/resources/recovery-configurations.test.js test/unit/models/recovery-configuration.test.js test/unit/models/recovery-configuration-transition.test.js test/unit/models/fsm-transition.test.js test/unit/models/pivtoken.test.js test/unit/models/pivtoken-history.test.js test/unit/models/recovery-token.test.js test/unit/helpers.js test/mockcloud-pivtokens.js test/integration/cli-pivtoken.test.js test/integration/cli-recovery.test.js test/integration/cli-subcommands.test.js test/integration/helpers.js test/run.js test/lib/client.js test/lib/server.js test/lib/log.js test/lib/config.js test/lib/token.js test/lib/mocks.js test/lib/common.js
json --validate -f package.json
deps/eng/tools/check-copyright -q
check ok
thanagar: triton-kbmapi(TRITON-2068) pedropc$ git checkout -- Makefile
thanagar: triton-kbmapi(TRITON-2068) pedropc$ make check
xmllint --noout --path deps/eng/tools/ --dtdvalid deps/eng/tools/service_bundle.dtd.1 smf/manifests/kbmapi.xml
xmllint --noout --path deps/eng/tools/ --dtdvalid deps/eng/tools/service_bundle.dtd.1 smf/manifests/kbmtr.xml
node node_modules/.bin/eslint  lib/apis/moray.js lib/endpoints/pivtokens.js lib/endpoints/index.js lib/endpoints/recovery-configurations.js lib/util/constants.js lib/util/errors.js lib/util/validate.js lib/auth.js lib/models/recovery-token.js lib/models/fsm-transition.js lib/models/recovery-configuration-transition.js lib/models/recovery-configuration.js lib/models/index.js lib/models/pivtoken.js lib/models/pivtoken-history.js lib/models/model.js lib/config.js lib/cli/do_completion.js lib/cli/do_pivtoken/do_add.js lib/cli/do_pivtoken/index.js lib/cli/do_pivtoken/do_remove.js lib/cli/do_pivtoken/do_list.js lib/cli/index.js lib/cli/do_recovery/do_activate.js lib/cli/do_recovery/do_add.js lib/cli/do_recovery/index.js lib/cli/do_recovery/do_remove.js lib/cli/do_recovery/do_get.js lib/cli/do_recovery/do_list.js lib/cli/do_recovery/do_stage.js lib/cli/do_recovery/do_cancel.js lib/cli/common.js lib/app.js client/index.js test/unit/resources/pivtokens.test.js test/unit/resources/recovery-configurations.test.js test/unit/models/recovery-configuration.test.js test/unit/models/recovery-configuration-transition.test.js test/unit/models/fsm-transition.test.js test/unit/models/pivtoken.test.js test/unit/models/pivtoken-history.test.js test/unit/models/recovery-token.test.js test/unit/helpers.js test/mockcloud-pivtokens.js test/integration/cli-pivtoken.test.js test/integration/cli-recovery.test.js test/integration/cli-subcommands.test.js test/integration/helpers.js test/run.js test/lib/client.js test/lib/server.js test/lib/log.js test/lib/config.js test/lib/token.js test/lib/mocks.js test/lib/common.js
json --validate -f package.json
deps/eng/tools/check-copyright -q
check ok
```